doc/ufunc_examples.rst

```diff
@@ -0,0 +1,192 @@
+
+    from xgcm import as_grid_ufunc
+
+
+In two dimensions, the divergence operator accepts two vector components and returns one scalar result.
+Each vector component will be differentiated along one axis, and doing so with a first order forward difference would
+shift the data's position along that axis.
+Therefore our signature should look something like this ``"(X:center,Y:center),(X:center,Y:center)->(X:center,Y:center)"``.
+
+A divergence is the sum of multiple partial derivatives, so first let's define a derivative function like this
+
+.. ipython:: python
+
+    def diff_1d(a):
+        return 0.5 * (a[..., 2:] - a[..., :-2])
+
+    def diff_center_to_center_second_order(arr, axis):
+        return np.apply_along_axis(diff_1d, axis, arr)
+
+Now if we treat the components of the ``(U, V)`` vector as independent scalars, our grid ufunc could be defined like this
+
+.. ipython:: python
+
+    @as_grid_ufunc("(X:center,Y:center),(X:center,Y:center)->(X:center,Y:center)", boundary_width={'X': (2, 0), 'Y': (2, 0)})
+    def divergence(u, v):
+        u_diff_x = diff_center_to_center_second_order(u, axis=-2)
+        v_diff_y = diff_center_to_center_second_order(v, axis=-1)
+        # Need to trim off elements so that the two arrays have same shape
+        div = u_diff_x[..., :-2] + v_diff_y[..., :-2, :]
+        return div
+
+Now we can compute the divergence of our example vector field
+
+.. ipython:: python
+
+    div = divergence(grid, ds['U'], ds['V'], axis=[('X', 'Y'), ('X', 'Y')])
+
+We can see the result lies on the expected coordinate positions
+
+.. ipython:: python
+
+    div.coords
+
+and the resulting divergence looks like it corresponds with the arrows of the vector field above
+
+.. ipython:: python
+
+    @savefig div_vector_field.png width=4in
+    div.plot(x='x_c', y='y_c')
+
+
+
+Gradient
+~~~~~~~~
+
+The gradient is almost like the opposite of divergence in the sense that it accepts one scalar and returns multiple vectors.
+
+For this lets first create a scalar field by computing the magnitude of our vector field
+
+.. ipython:: python
+
+    a = U**2 + V**2
+    ds['a'] = a
+
+    @savefig scalar_field.png width=4in
+    a.plot(x='x_c')
+
+
+The signature for this ufunc will look like this ``"(X:center,Y:center)->(X:center,Y:center),(X:center,Y:center)"``,
+and our definition is similar to the derivative case.
+
+.. ipython:: python
+    :okexcept:
+
+    @as_grid_ufunc("(X:center,Y:center)->(X:center,Y:center),(X:center,Y:center)", boundary_width={'X': (2, 0), 'Y': (2, 0)})
+    def gradient(a):
+        a_diff_x = diff_center_to_center_second_order(a, axis=-2)
+        a_diff_y = diff_center_to_center_second_order(a, axis=-1)
+        # Need to trim off elements so that the two arrays have same shape
+        return a_diff_x[..., :-2], a_diff_y[..., :-2, :]
+
+Now we can compute the gradient of our example scalar field
+
+.. ipython:: python
+
+    ds['grad_x'], ds['grad_y'] = gradient(grid, ds['a'], axis=[('X', 'Y')])
```

Comment:
I kind of prefer the syntax 
```suggestion
    ds['grad_x'], ds['grad_y'] = grid.apply_grid_ufunc(gradent, ds['a'], axis=[('X', 'Y')])
```

Is that valid?

Reply:
The method is called `apply_as_grid_ufunc` (I can't decide whether I like the "as"'s) but yes that's valid.

Reply:
And yes we can mix both syntaxes, though I don't really see the point of doing that when it's already clearly explained in #431